src/index.ts

```diff
@@ -1,89 +1,148 @@
-import { Kafka, logLevel } from "kafkajs";
-import { Entry, FhirMapping, FhirPlugin, Table } from "./types";
-import { GetFhirPlugins, GetTableMappings, ValidateFhirMappingsJson } from "./util";
-import { loadDataIntoClickhouse, selectByIdClickhouse, alterRowIntoClickhouse } from "./clickhouse/utils";
-
-const fhirMappings: FhirMapping[] = require("./data/fhir-mapping.json");
-const fhirMappingValidationErrors = ValidateFhirMappingsJson(fhirMappings);
-if (fhirMappingValidationErrors.length > 0) {
-  console.error("Invalid fhir-mapping.json file");
-  fhirMappingValidationErrors.forEach((error) => console.error(error));
-  process.exit(1);
-}
+import {Kafka, logLevel} from 'kafkajs'
+import {
+  registerMediator,
+  activateHeartbeat,
+  fetchConfig
+} from 'openhim-mediator-utils'
+import {promisify} from 'util'
+import {Entry, FhirMapping, FhirPlugin, Table} from './types'
+import {
+  GetFhirPlugins,
+  GetTableMappings,
+  ValidateFhirMappingsJson
+} from './util'
+import {
+  loadDataIntoClickhouse,
+  selectByIdClickhouse,
+  alterRowIntoClickhouse
+} from './clickhouse/utils'
+import mediatorConfig from './data/mediator-config.json'
+
+let fhirMappings: FhirMapping[] = require('./data/fhir-mapping.json')
+// Set default mediator FHIR mapping from config file
+mediatorConfig.config.fhirMappings = JSON.stringify(fhirMappings, null, 2)
```

Comment:
I may be wrong, I wonder why we need to stringify the fhirMapping. Also, I saw that in openHIM there's the concept of variables, I wonder we re-use of that ?

Reply:
The data type that I'm using for storing the config in the OpenHIM is a simple string. So, this has to be converted to a string here else the validation will complain in the OpenHIM config API.